boot/zephyr/keys.c

```diff
@@ -41,7 +41,7 @@ extern unsigned int ecdsa_pub_key_len;
 #define HAVE_KEYS
 extern const unsigned char ed25519_pub_key[];
 extern unsigned int ed25519_pub_key_len;
-#else
+#elif !defined(CONFIG_BOOT_SIGNATURE_TYPE_NONE)
```

Comment:
Wouldn't it be better to just remove the `#else` and the `#error`. The NONE variant is more an issue of how the configuration works with Zephyr, but it seems reasonable to just allow none of them to be defined, rather than requiring the NONE variant.

Reply:
Reasonable. I'll make the change.